In [33]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import sklearn
import scipy


In [34]:
df = pd.read_csv('data.csv')

In [35]:
df.head()


date      price  bedrooms  bathrooms  sqft_living  sqft_lot  \
0  2014-05-02 00:00:00   313000.0       3.0       1.50         1340      7912   
1  2014-05-02 00:00:00  2384000.0       5.0       2.50         3650      9050   
2  2014-05-02 00:00:00   342000.0       3.0       2.00         1930     11947   
3  2014-05-02 00:00:00   420000.0       3.0       2.25         2000      8030   
4  2014-05-02 00:00:00   550000.0       4.0       2.50         1940     10500   

   floors  waterfront  view  condition  sqft_above  sqft_basement  yr_built  \
0     1.5           0     0          3        1340              0      1955   
1     2.0           0     4          5        3370            280      1921   
2     1.0           0     0          4        1930              0      1966   
3     1.0           0     0          4        1000           1000      1963   
4     1.0           0     0          4        1140            800      1976   

   yr_renovated                    street       city  statezip country  
0          2005      18810 Densmore Ave N  Shoreline  WA 98133     USA  
1             0           709 W Blaine St    Seattle  WA 98119     USA  
2             0  26206-26214 143rd Ave SE       Kent  WA 98042     USA  
3             0           857 170th Pl NE   Bellevue  WA 98008     USA  
4          1992         9105 170th Ave NE    Redmond  WA 98052     USA

In [36]:
df.shape

(4600, 18)

In [37]:
# Identify zero values in the dataset
zero_values = df[df == 0]

# Count the number of zero values in each column
zero_counts = zero_values.count()

# Print columns with zero values and their counts
print("The zero values:")
print(zero_counts[zero_counts > 0])

The zero values:
price              49
bedrooms            2
bathrooms           2
waterfront       4567
view             4140
sqft_basement    2745
yr_renovated     2735
dtype: int64


 From be above  we note that, "price" has 49 zeros, "bedrooms" has 2, "bathrooms" has 2, "waterfront" has 4567, "view" has 4140, "sqft_basement" has 2745, and "yr_renovated" has 2735.
 we will have to drop the row and columns

In [38]:


# Drop the specified columns
df2 = df.drop(['waterfront', 'view', 'sqft_basement', 'yr_renovated','sqft_basement','yr_renovated','statezip','country'], axis='columns')
df2.head()


date      price  bedrooms  bathrooms  sqft_living  sqft_lot  \
0  2014-05-02 00:00:00   313000.0       3.0       1.50         1340      7912   
1  2014-05-02 00:00:00  2384000.0       5.0       2.50         3650      9050   
2  2014-05-02 00:00:00   342000.0       3.0       2.00         1930     11947   
3  2014-05-02 00:00:00   420000.0       3.0       2.25         2000      8030   
4  2014-05-02 00:00:00   550000.0       4.0       2.50         1940     10500   

   floors  condition  sqft_above  yr_built                    street  \
0     1.5          3        1340      1955      18810 Densmore Ave N   
1     2.0          5        3370      1921           709 W Blaine St   
2     1.0          4        1930      1966  26206-26214 143rd Ave SE   
3     1.0          4        1000      1963           857 170th Pl NE   
4     1.0          4        1140      1976         9105 170th Ave NE   

        city  
0  Shoreline  
1    Seattle  
2       Kent  
3   Bellevue  
4    Redmond

In [39]:
# Display information about the DataFrame
df2.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4600 entries, 0 to 4599
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   date         4600 non-null   object 
 1   price        4600 non-null   float64
 2   bedrooms     4600 non-null   float64
 3   bathrooms    4600 non-null   float64
 4   sqft_living  4600 non-null   int64  
 5   sqft_lot     4600 non-null   int64  
 6   floors       4600 non-null   float64
 7   condition    4600 non-null   int64  
 8   sqft_above   4600 non-null   int64  
 9   yr_built     4600 non-null   int64  
 10  street       4600 non-null   object 
 11  city         4600 non-null   object 
dtypes: float64(4), int64(5), object(3)
memory usage: 431.4+ KB


In [40]:
# Describe the data in the DataFrame
description = df2.describe()

# Display the description
print(description)

              price     bedrooms    bathrooms   sqft_living      sqft_lot  \
count  4.600000e+03  4600.000000  4600.000000   4600.000000  4.600000e+03   
mean   5.519630e+05     3.400870     2.160815   2139.346957  1.485252e+04   
std    5.638347e+05     0.908848     0.783781    963.206916  3.588444e+04   
min    0.000000e+00     0.000000     0.000000    370.000000  6.380000e+02   
25%    3.228750e+05     3.000000     1.750000   1460.000000  5.000750e+03   
50%    4.609435e+05     3.000000     2.250000   1980.000000  7.683000e+03   
75%    6.549625e+05     4.000000     2.500000   2620.000000  1.100125e+04   
max    2.659000e+07     9.000000     8.000000  13540.000000  1.074218e+06   

            floors    condition   sqft_above     yr_built  
count  4600.000000  4600.000000  4600.000000  4600.000000  
mean      1.512065     3.451739  1827.265435  1970.786304  
std       0.538288     0.677230   862.168977    29.731848  
min       1.000000     1.000000   370.000000  1900.000000  
25%   